### PR TITLE
users trigger wall

### PR DESCRIPTION
**Pull request by jonybravo**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,4 +3,8 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+  # Associations
+  has_one :wall, dependent: :destroy
+  before_create :build_wall
```
Comment (andreorvalho) [inline app/models/user.rb]: do you know what before_create is? and how it works?
http://guides.rubyonrails.org/active_record_callbacks.html

**Comment by andreorvalho** [comment]
here it explains how associations works:
http://guides.rubyonrails.org/association_basics.html

your design/code has almost everything correct. The reason why this is not working, is because you forgot to do a migration. If you don't change the database how do you think a user and a wall relate?

Attention, here nothing triggers nothing. A user has a wall, and a wall belongs to a user. So that means that every time a new user is created a wall should be created as well. You have different ways to do that.

You can also watch this video here: http://ruby-on-rails.wonderhowto.com/how-to/use-one-one-association-when-using-ruby-rails-3-407824/

try to implement it and let me know if you have doubts.

**Comment by jonybravo** [comment]
screen shot 2015-11-28 at 17 12 49

I don't see why this error came up!

**Comment by andreorvalho** [comment]
The answer to your problem is here: http://stackoverflow.com/questions/5547730/rails-factory-girl-getting-email-has-already-been-taken (read the second answer as the accepted answer is outdated)


Basically devise has a validation checking that 2 users cannot have the same email(smart huh?) and because of that when you create a factory that does not create users with different emails. Every time you produce the first user everything is fine, but if you produce the second user it will try to create a record with the same email and fail.

**Comment by andreorvalho** [comment]
Oh I just realised you figured it out :) good job.

**Comment by andreorvalho** [comment]
I will merge since everything seems fine.

**Pull request merged**